sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/DigitalTwinModelsRestClient.cs

```diff
@@ -21,37 +21,27 @@ internal partial class DigitalTwinModelsRestClient
             CreateModelsOptions digitalTwinModelsAddOptions = null,
             CancellationToken cancellationToken = default)
         {
-            using DiagnosticScope scope = _clientDiagnostics.CreateScope("DigitalTwinModelsClient.Add");
-            scope.Start();
-            try
-            {
-                using HttpMessage message = CreateAddRequest(models, digitalTwinModelsAddOptions);
-                await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
-                switch (message.Response.Status)
-                {
-                    case 200:
-                    case 201:
+            using HttpMessage message = CreateAddRequest(models, digitalTwinModelsAddOptions);
+            await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
+            switch (message.Response.Status)
+            {
+                case 200:
+                case 201:
+                    {
+                        IReadOnlyList<DigitalTwinsModelData> value = default;
+                        using JsonDocument document = await JsonDocument
+                            .ParseAsync(message.Response.ContentStream, default, cancellationToken)
+                            .ConfigureAwait(false);
+                        List<DigitalTwinsModelData> array = new List<DigitalTwinsModelData>(document.RootElement.GetArrayLength());
+                        foreach (JsonElement item in document.RootElement.EnumerateArray())
                         {
-                            IReadOnlyList<DigitalTwinsModelData> value = default;
-                            using JsonDocument document = await JsonDocument
-                                .ParseAsync(message.Response.ContentStream, default, cancellationToken)
-                                .ConfigureAwait(false);
-                            List<DigitalTwinsModelData> array = new List<DigitalTwinsModelData>(document.RootElement.GetArrayLength());
-                            foreach (JsonElement item in document.RootElement.EnumerateArray())
-                            {
-                                array.Add(DigitalTwinsModelData.DeserializeDigitalTwinsModelData(item));
-                            }
-                            value = array;
-                            return Response.FromValue(value, message.Response);
+                            array.Add(DigitalTwinsModelData.DeserializeDigitalTwinsModelData(item));
                         }
-                    default:
-                        throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
-                }
-            }
-            catch (Exception ex)
```

Comment:
here and below - we don't need to catch and rethrow the exception anymore. The details are captured in the diagnostic scope in the convenience layer.

Reply:
Thanks @abhipsaMisra! Will merge this in once the CI turns green.